### PR TITLE
Update GreaterThan to handle nans

### LINKED ISSUE #589: Null values in GreaterThan constraint raises error
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version: Any
* Python version: Any
* Operating System: Any

### Error Description

If you create a `GreaterThan` constraint with columns that have null values, a `ConstraintsNotMetError` is raised.

### Steps to reproduce

You can use the `student_placements` demo dataset to recreate this error.

```
student_placements = load_tabular_demo('student_placements')
gt_constraint = GreaterThan(
    low='start_date',
    high='end_date',
    handling_strategy='reject_sampling'
)
gc = GaussianCopula(constraints=[gt_constraint])
gc.fit(student_placements)
```

Running the same code above after dropping all rows in `student_placements` where `start_date` is NaN causes the error to go away.

### LINKED ISSUE #590: ConstraintsNotMetError when using GreaterThan constraint with datetime
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version: 0.12.0
* Python version: 3.8.8
* Operating System: Windows

### Error Description

ConstraintsNotMetError is raised when trying to use GreaterThan constraint with datetime. 

### Steps to reproduce

I have included a working example using sdv demo dataset

```
from sdv.demo import load_tabular_demo
from sdv.constraints import GreaterThan

data = load_tabular_demo('student_placements')

start_constraint = GreaterThan(
    low='start_date',
    high='end_date',
    handling_strategy='reject_sampling' )
constraints = [start_constraint]

gc = GaussianCopula(constraints=constraints)
gc.fit(data)`
```
And the error obtained is pasted below 

```
ConstraintsNotMetError                    Traceback (most recent call last)
<ipython-input-38-bffc85f02fc5> in <module>
     11 
     12 gc = GaussianCopula(constraints=constraints)
---> 13 gc.fit(data)

~\Anaconda3\lib\site-packages\sdv\tabular\base.py in fit(self, data)
    136 
    137         LOGGER.debug('Transforming table %s; shape: %s', self._metadata.name, data.shape)
--> 138         transformed = self._metadata.transform(data)
    139 
    140         if self._metadata.get_dtypes(ids=False):

~\Anaconda3\lib\site-packages\sdv\metadata\table.py in transform(self, data, on_missing_column)
    620         data = self._anonymize(data[fields])
    621 
--> 622         self._validate_data_on_constraints(data)
    623 
    624         LOGGER.debug('Transforming constraints for table %s', self.name)

~\Anaconda3\lib\site-packages\sdv\metadata\table.py in _validate_data_on_constraints(self, data)
    592             if set(constraint.constraint_columns).issubset(data.columns.values):
    593                 if not constraint.is_valid(data).all():
--> 594                     raise ConstraintsNotMetError('Data is not valid for the given constraints')
    595 
    596     def transform(self, data, on_missing_column='error'):

ConstraintsNotMetError: Data is not valid for the given constraints```

### LINKED ISSUE #592: `GreaterThan` constraint crashing with `NaN` values
As reported in issue #590 and #589, the `GreaterThan` seems to be failing in the presence of NaN values. The cause seems to be the following line:

https://github.com/sdv-dev/SDV/blob/061d69b93f8d4ca8b974da158bc76b67c44b00eb/sdv/metadata/table.py#L593

Basically, since `NaN` values are not comparable, this line is returning `false` for the comparison between `NaN`'s, which causes the code to crash.

We should work on a fix, seeing as this error happens with our demo dataset (student_placements).

### PR DESCRIPTION
**Pull request by katxiao**
Resolves #592 
Resolves #589 
Resolves #590

### CODE REVIEW

File: sdv/constraints/tabular.py [hunk 1]
```diff
@@ -442,8 +442,11 @@ def is_valid(self, table_data):
         """
         low = self._get_value(table_data, 'low')
         high = self._get_value(table_data, 'high')
+        isnull = np.logical_or(np.isnan(low), np.isnan(high))
 
-        return self.operator(high, low).all(axis=1)
+        valid = np.logical_or(self.operator(high, low), isnull)
```
Comment (amontanez24) [inline sdv/constraints/tabular.py:447]: Too bad there is no way for the operator to ignore nulls.

**Review by amontanez24 — APPROVED** [review]
Looks good!

**Review by csala — APPROVED** [review]
LGTM!

File: sdv/constraints/tabular.py (hunk 1 above)
Comment (katxiao) [reply sdv/constraints/tabular.py:447]: Yeah, it would be nice if there was a built-in option to ignore nulls.

**Pull request merged**